internal/pkg/service/stream/keboolasink/bridge/file.go

```diff
@@ -103,9 +116,70 @@ func (b *Bridge) deleteCredentialsOnFileDelete() {
 	b.plugins.Collection().OnFileDelete(func(ctx context.Context, now time.Time, original, file *model.File) error {
 		if b.isKeboolaStagingFile(file) {
 			op.AtomicOpFromCtx(ctx).Write(func(ctx context.Context) op.Op {
-				return b.schema.UploadCredentials().ForFile(file.FileKey).Delete(b.client)
+				return b.schema.File().ForFile(file.FileKey).Delete(b.client)
 			})
 		}
 		return nil
 	})
 }
+
+func (b *Bridge) registerFileImporter() {
+	b.plugins.RegisterFileImporter(
+		targetProvider,
+		func(ctx context.Context, file *plugin.File) error {
+			// Get authorization token
+			var token *keboolasink.Token
+			err := b.schema.Token().ForSink(file.SinkKey).GetOrNil(b.client).WithResultTo(&token).Do(ctx).Err()
+			if err != nil {
+				return err
+			}
+
+			// Get file upload credentials
+			var keboolaFile *keboolasink.File
+			err = b.schema.File().ForFile(file.FileKey).GetOrNil(b.client).WithResultTo(&keboolaFile).Do(ctx).Err()
+			if err != nil {
+				return err
+			}
+
+			if keboolaFile == nil {
+				return errors.New("missing keboola details for file")
+			}
+
```

Comment:
`GetOrNil` -> `GetOrErr` (we want to rename it to `GetMustExist`) and you can omit the `if keboolaFile == nil ` condition.